woocommerce/woocommerce-gutenberg-products-block
# Update/dynamic grid blocks html

This is related to #558. I originally planned to target #565 but I'm not 100% sure if thats going ahead so I switched to the other branch.

This PR updates the HTML rendering for products so that:

- No core template functions are used
- Our HTML pretty much only comes from a product object
- CSS handles visibility of elements like in the editor
- The `woocommerce` div wrapper is added back since themes often rely on this for styling. e.g. the sale badges were out of position without this.

There was also the `__CLASS__` issue I pointed out in a previous review which should now be corrected.

### Accessibility

This PR changes HTML but no existing styles or interactions.

### Screenshots

**Styling still needs work** on these dynamic block PRs, but this shows before:

![Test – local wordpress test 2019-05-15 15-41-32](https://user-images.githubusercontent.com/90977/57784511-ed4a2500-7727-11e9-90ac-609b4077e37e.png)

And after:

![Test – local wordpress test 2019-05-15 15-37-42](https://user-images.githubusercontent.com/90977/57784182-639a5780-7727-11e9-9489-b015f4178776.png)

The sale badges work now because of the wrapping woocommerce div.

### How to test the changes in this Pull Request:

1. Checkout and build
2. View the new/on-sale product grid blocks in the editor and on the frontend
3. Styling and appearance should match shortcodes. Viewing source/inspecting you should see the new HTML markup, including inline classes for the options such as `is-hidden-title` if applicable
4. You should also see a div around the block with class `woocommerce`
